solr/core/src/java/org/apache/solr/jersey/SolrJacksonMapper.java

```diff
@@ -18,15 +18,44 @@
 package org.apache.solr.jersey;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import java.io.IOException;
 import javax.ws.rs.ext.ContextResolver;
 import javax.ws.rs.ext.Provider;
+import org.apache.solr.common.util.NamedList;
 
 /** Customizes the ObjectMapper settings used for serialization/deserialization in Jersey */
+@SuppressWarnings("rawtypes")
 @Provider
 public class SolrJacksonMapper implements ContextResolver<ObjectMapper> {
   @Override
   public ObjectMapper getContext(Class<?> type) {
-    return new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    final SimpleModule customTypeModule = new SimpleModule();
+    customTypeModule.addSerializer(new NamedListSerializer(NamedList.class));
+
+    return new ObjectMapper()
+        .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+        .registerModule(customTypeModule);
+  }
+
+  public static class NamedListSerializer extends StdSerializer<NamedList> {
+
+    public NamedListSerializer() {
+      this(null);
+    }
+
+    public NamedListSerializer(Class<NamedList> nlClazz) {
+      super(nlClazz);
+    }
+
+    @Override
+    public void serialize(NamedList value, JsonGenerator gen, SerializerProvider provider)
+        throws IOException {
+      gen.writeObject(value.asShallowMap());
```

Comment:
I suspect if benchmarks are done, this is going to be revisited to be more efficient.  That NamedList.shallowMap looks trivial but the implementations of several of its methods call NamedList.asMap which is not.  Fine for now I guess.
CC @noblepaul